raiden/raiden_service.py

```diff
@@ -341,9 +340,13 @@ def stop(self):
         # We need a timeout to prevent an endless loop from trying to
         # contact the disconnected client
         try:
-            with gevent.Timeout(self.shutdown_timeout):
+            with gevent.Timeout(self.shutdown_timeout, 'Substasks didn\'t stopped on time'):
+                self.transport.stop()
+                self.alarm.stop()
+                self.transport.get()
```

Comment:
`stop` isn't intended to raise run-time exceptions, only stop-time ones (as well as `start` raise startup-time exceptions). This `get` here ensures run-time exceptions are re-raised, if any.

Reply:
I don't understand what you mean by runtime/stoptime/cleanup exceptions.

Reply:
Imagine the life of a Runnable: `start` -> `run` -> `stop`
- `start` is synchorous, and may raise if something **during** startup didn't go as expected. Think on a port binding when the port is already in use, and it should prevent the task from being actually started (as some pre-start condition failed)
- `runtime` exceptions are raised _during_ "normal" operation, inside `run` blocking call, either by itself (in case its a loop) or by any subtask (through `kill` or `get`)
- `stop` should also be synchronous, and it should fail on **cleanup** exceptions (imagine a `commit` on a DB which failed). No need to re-raise run-time exceptions

Reply:
The call to `stop` then `get` still don't make sense.

If you want to call `get` because there could have been a "runtime" exception, then the task is dead already and calling `stop` doesn't make sense.

Reply:
Runtime exceptions **may** be raised **because** of `stop`. It shouldn't be common, but can happen that `stop` tells the greenlet to exit gracefully, but then an exception is raised inside it. The interface still holds.